app/code/community/Drip/Connect/Model/Observer/Customer/GuestSubscriberCreated.php

```diff
@@ -9,15 +9,26 @@ class Drip_Connect_Model_Observer_Customer_GuestSubscriberCreated extends Drip_C
      */
     protected function executeWhenEnabled($observer)
     {
-        if (! Mage::registry(self::REGISTRY_KEY_NEW_GUEST_SUBSCRIBER)) {
-            return;
-        }
-
         $email = Mage::app()->getRequest()->getParam('email');
+
         $subscriber = Mage::getModel('newsletter/subscriber')->loadByEmail($email);
 
         $config = Drip_Connect_Model_Configuration::forCurrentScope();
 
+        if (! Mage::registry(self::REGISTRY_KEY_NEW_GUEST_SUBSCRIBER)) {
```

Comment:
My understanding was that when this case is hit, we invoke the `\Drip\Connect\Observer\Customer\AfterSave` observer. Is that not firing?